adapters/sinatra/lib/opentelemetry/adapters/sinatra/adapter.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'sinatra'
+require 'opentelemetry'
+
+require_relative 'extensions/tracer_extension'
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      class Adapter
+        class << self
+          attr_reader :config,
+                      :propagator
+
+          def install(config = {})
+            return :already_installed if installed?
+
+            @config = config
+            @propagator = tracer_factory.http_text_format
+
+            new.install
+          end
+
+          def tracer
+            @tracer ||= OpenTelemetry.tracer_factory.tracer(
+              Sinatra.name,
+              Sinatra.version
+            )
+          end
+
+          attr_accessor :installed
+          alias_method :installed?, :installed
+
+          private
+
+          def tracer_factory
```

Comment:
This method seems unnecessary. It is only called in one place (`install`) and is *not* used in one other place it could be (`tracer`). I don't think it provides enough value.

Reply:
removed :heavy_check_mark: